tests/OptionsTest.php

```diff
@@ -519,4 +538,34 @@ public function testReleaseOptionDefaultValueIsGotFromEnvironmentVariable(): voi
 
         $this->assertSame('0.0.1', (new Options())->getRelease());
     }
+
+    public function testErrorTypesIsDynamiclyReadFromErrorReportingLevelWhenUnset(): void
```

Comment:
I don't understand the usefulness of neither this nor the next test. As far as I understand from the changes, it should be enough to test just these cases (using a data provider):

- the getter returns `E_*` when `error_types = E_*`
- the getter returns the current value of `error_reporting()` when `error_types = null`

All the code to test that when changing `error_reporting()` the value obtained from the getter is the right one is made superfluous by the first test case above, isn't it?

Reply:
The difference is that the value is not statically saved anywhere, but it's fetched from `error_reporting()` each time.

Reply:
Yeah that was the intention of the 2 tests to validate that in the first when `error_types` is `null` it changes depending on the current value of `error_reporting` and in the second it doesn't matter what `error_reporting()` says it's always the same.

I thought it be easier to read like this too without a data provider since we need to modify the `error_reporting()` to make sure we see the "live" value in the first case.

But feel free to refactor it to something else if that makes more sense.

Reply:
>The difference is that the value is not statically saved anywhere, but it's fetched from error_reporting() each time.

But this is asserted in the second case I mentioned. Let's try to rewrite the following code to analyze the possible outputs:

https://github.com/getsentry/sentry-php/blob/41bf1efb217f1fa98658d59104505b28ea63e314/src/Options.php#L414

It becomes:

```php
if ($this->options['error_types'] !== null) {
    return $this->options['error_types'];
}

return error_reporting();
```

There are just two exit points from the function, and they are both covered with the test cases mentioned above. There is no point in checking here that if you set a different value for the `error_reporting` then the new value is reported because, well, there is no other way it could not 😃

Reply:
>It does if you consider the previous behavior right?

Yes, I think that what you say makes sense. But in turn this is tested in the PHPT which is a real integration test, isn't it?